docs/merkletree/treetex/main.go

```diff
@@ -96,13 +99,14 @@ var (
 
 // nodeInfo represents the style to be applied to a tree node.
 type nodeInfo struct {
-	incProof     bool
-	incPath      bool
-	target       bool
-	perfectRoot  bool
-	ephemeral    bool
-	leaf         bool
-	rangeIndices []int
+	proof             bool
+	incPath           bool
+	target            bool
+	perfectRoot       bool
+	ephemeral         bool
+	leaf              bool
+	rangeIndices      []int
+	proofRangeIndices []int
```

Comment:
Why do you distinguish `rangeIndices` and `proofRangeIndices`? As I understand it, the source of confusion is that the leaf nodes are in fact storing 2 nodes, so one array is used by "upper" leaf nodes, and the other one - by "lower" ones.

Consider separating "data node info" to make this division explicit.

Reply:
Yup, that's right.
I'll add a TODO for that.